### ISSUE TITLE
subplot duplication in overlayed axis

### ISSUE THREAD

**Issue by vinipitta** [plotly/plotly.js#2347]
Scenario: I have a plot with two y axis overlayed where trace data is loaded asynchronously so there is no guarantee  of order. When a trace that belongs to the y axis overlaying the main y axis is loaded first, the plot generate 2 set o graphic elements on SVG representing the trace: the first as a subplot and the second as an overplot of the mainplot.

Here is a live example: https://codepen.io/vinipitta/pen/WMGdbx
(Just drag the defective trace (trace 1) to see the duplicated trace remain still while a copy moves on the right y axis.)

Looks like since there is no trace loaded in the main y axis, the _plotinfo.mainplot_ is **undefined** when the trace is added to the overlayed axis generating a subplot. When a trace is added to the mainplot, the overplot group is populated with the previous trace and the former subplot ins't removed.

```
function makeSubplotLayer(plotinfo) {
    var plotgroup = plotinfo.plotgroup;
    var id = plotinfo.id;
    var xLayer = constants.layerValue2layerClass[plotinfo.xaxis.layer];
    var yLayer = constants.layerValue2layerClass[plotinfo.yaxis.layer];

    if(!plotinfo.mainplot) {

```
The simplest workaround is to set the _layout.yaxis.overlaying_ or _layout.yaxis2.overlaying_ based on where the first loaded trace belongs to.

This issue seems related to: #2227

**Comment by alexcjohnson** [comment]
Thanks for the report. This specific manifestation indeed came in with #2227 though I suspect you could have caused the same failure in other ways before that.

**Comment by etpinard** [comment]
This seems to be fixed using 1.41.3

![image](https://user-images.githubusercontent.com/6675409/46498387-4c87a480-c7eb-11e8-8534-649453220c1e.png)

----

Related: https://github.com/plotly/plotly.js/issues/3074

**Comment by alexcjohnson** [comment]
Seems to have been fixed in fact in v1.39.3 - presumably by #2831 - 🏆 @etpinard
